blog/2024-02-09-private-api-gateway-aws.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Building a Private API Gateway in AWS with Winglang
+description: A step by step guid to build a private API Gateway in an AWS VPC with Winglang
+authors:
+  - skorfmann
+  - hasanaburayyan
+tags: [aws, api-gateway, vpc]
+hide_table_of_contents: false
+---
+
+This tutorial is a guide for creating a [secure](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-private-apis.html) [API Gateway](https://aws.amazon.com/api-gateway/) inside an AWS Virtual Private Cloud (VPC) using Winglang. A private API Gateway in AWS is used when you need to expose APIs that are only accessible within your Virtual Private Cloud (VPC) or through specific endpoints that you define. This setup is essential for internal or private applications where security and controlled access are crucial.
+
+Wing currently supports two programming languages: TypeScript and [Winglang](https://winglang.io). For the purposes of this tutorial, we'll use Winglang.
+
+Let's start by installing the Wing CLI (you'll need Node.js >= 18.x installed):
+
+```bash
+npm i -g winglang
+```
+
+You can check the CLI version like this:
+
+```bash
+wing --version
+0.57.22
+```
+
+> As you can see, Wing is still in pre-release, so expect some hiccups and don't be shy reporting issues or get help from the awesome people hanging out on the [Wing slack](https://t.winglang.io/slack).
+>
+
+Ok, now that we have the Wing CLI installed, let's create a new project using the `private-api` quickstart:
+
+```bash
+$ mkdir my-api
+$ cd my-api
+$ wing new private-api
```

Comment:
@Chriscbr my immediate reaction from this output was that "something didn't work for me" 

![image](https://github.com/winglang/docsite/assets/1727147/285a7c2e-548d-4ceb-b5af-e30c89e71ba4)

Maybe we should be less verbose or maybe it is only me...

Reply:
It's not only you @ekeren . To me the use of the yellow and red colors in the text looked like there were errors and warnings in the execution. It also felt too verbose and with too many warnings

Reply:
Great, I am glad I mentioned it then

Reply:
@Chriscbr is it something you can address please? I think that less verbose will help here.

Reply:
@ekeren @ShaiBer Not sure I understand - I could be totally misguided here but there aren't any yellow or red colors in the text...?

2 cents - when I've tried new tools and I want to get started immediately, I've found it incredibly helpful when they point out the most common commands (not requiring me to dig through docs or man pages). I'm not sure if there's huge value in subtracting text when the command is on the path for new users and it's seldom run outside of that context. Here are two examples I took from other tools:

**react:**
<img width="537" alt="Screenshot 2024-02-19 at 3 34 42 PM" src="https://github.com/winglang/docsite/assets/5008987/6077e93f-2d82-433b-af2b-98e12cb033ad">

**astro:**
<img width="464" alt="Screenshot 2024-02-19 at 6 14 31 PM" src="https://github.com/winglang/docsite/assets/5008987/ab3e87f0-40f4-4b00-9cdd-8720655a8444">

Is there something visually that would make it seem less error-like? e.g. adding "Success!" at the beginning?